apis/datadoghq/v2alpha1/datadogagent_types.go

```diff
@@ -38,9 +38,155 @@ type DatadogAgentSpec struct {
 }
 
 // DatadogFeatures are Features running on the Agent and Cluster Agent.
-// TODO add the features (APM, Kubernetes State Metrics Core...)
 // +k8s:openapi-gen=true
-type DatadogFeatures struct{}
+type DatadogFeatures struct {
+	// Application-level features
+	// APM (Application Performance Monitoring) configuration
+	APM *APMFeatureConfig `json:"apm,omitempty"`
+	// Logs configuration
+	Logs *LogsFeatureConfig `json:"logs,omitempty"`
+	// Process configuration
+	Process *ProcessFeatureConfig `json:"process,omitempty"`
+	// CWS (Cloud Workload Security) configuration
+	CWS *CWSFeatureConfig `json:"cws,omitempty"`
+	// CWS (Cloud Security Posture Management) configuration
+	CSPM *CSPMFeatureConfig `json:"cspm,omitempty"`
+	// NPM (NetworkPerformanceMonitoring) configuration
+	NPM *NPMFeatureConfig `json:"npm,omitempty"`
+
+	// Cluster-level features
+	// OrchestratorExplorer check configuration
+	OrchestratorExplorer *OrchestratorExplorerFeatureConfig `json:"orchestratorExplorer,omitempty"`
+	// KubeStateMetricsCore check configuration
+	KubeStateMetricsCore *KubeStateMetricsCoreFeatureConfig `json:"kubeStateMetricsCore,omitempty"`
+	// AdmissionController configuration
+	AdmissionController *AdmissionControllerFeatureConfig `json:"admissionController,omitempty"`
+	// ExternalMetricsServer configuration
+	ExternalMetricsServer *ExternalMetricsServerFeatureConfig `json:"externalMetricsServer,omitempty"`
+	// ClusterChecksRunner configuration
+	ClusterChecksRunner *ClusterChecksRunnerFeatureConfig `json:"clusterChecksRunner,omitempty"`
+	// DatadogMonitor configuration
+	DatadogMonitor *DatadogMonitorFeatureConfig `json:"datadogMonitor,omitempty"`
+}
+
+// Configuration structs for each feature in DatadogFeatures. These options are limited to the most
+// commonly configured parameters; other parameters will use default values unless specified in the spec.
+// Note that any conflicting configuration in the DatadogAgentSpec will take precedence.
+
+// APMFeatureConfig contains APM (Application Performance Monitoring) configuration.
+// APM runs in the Trace Agent
+type APMFeatureConfig struct {
+	// Enabled enables Application Performance Monitoring
+	// +optional
+	Enabled *bool `json:"enabled,omitempty"`
+	// HostPort takes a port number (0 < x < 65536) to expose on the host. (Most containers do not need this.)
+	// If HostNetwork is enabled, this value must match the ContainerPort
+	// +optional
+	HostPort *int32 `json:"hostPort,omitempty"`
+}
+
+// LogsFeatureConfig contains Logs configuration.
+// Logs collection is run in the Agent
+type LogsFeatureConfig struct {
```

Comment:
IMO all the [options](https://github.com/DataDog/datadog-operator/blob/5d59e5faa9454a984afdae533b0a5e8303fd1455/apis/datadoghq/v1alpha1/datadogagent_types.go#L374) that was present in v1alpha1 should also be present in this version